pandas/core/indexes/base.py

```diff
@@ -1833,13 +1833,14 @@ def _assert_take_fillable(self, values, indices, allow_fill=True,
                               fill_value=None, na_value=np.nan):
         """ Internal method to handle NA filling of take """
         indices = _ensure_platform_int(indices)
-
         # only fill if we are passing a non-None fill_value
         if allow_fill and fill_value is not None:
             if (indices < -1).any():
                 msg = ('When allow_fill=True and fill_value is not None, '
                        'all indices must be >= -1')
                 raise ValueError(msg)
+            if values.size == 0:
```

Comment:
Why only do this when `fill_value` is not None? Is that what we want? (I'm not sure). If so, maybe add a comment explaining it.